test/e2e/singlecluster/visibility_test.go

```diff
@@ -124,12 +124,19 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
 			})
 		})
 		ginkgo.AfterEach(func() {
-			gomega.Expect(k8sClient.Delete(ctx, lowPriorityClass)).To(gomega.Succeed())
-			gomega.Expect(k8sClient.Delete(ctx, midPriorityClass)).To(gomega.Succeed())
-			gomega.Expect(k8sClient.Delete(ctx, highPriorityClass)).To(gomega.Succeed())
-			gomega.Expect(util.DeleteObject(ctx, k8sClient, localQueueB)).Should(gomega.Succeed())
-			gomega.Expect(util.DeleteObject(ctx, k8sClient, localQueueA)).Should(gomega.Succeed())
 			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, nsA)).Should(gomega.Succeed())
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, nsB)).Should(gomega.Succeed())
```

Comment:
qq, a couple things are changed here:
- order of deletions, 
- added deletion of workloads
- retried delete for local queue and priorities

Do we actually know which of these changes is relevant for the fix?

Reply:
Yes, this one 

```go
// Force remove workloads to be sure that cluster queue can be removed.
			gomega.Expect(util.DeleteWorkloadsInNamespace(ctx, k8sClient, nsA)).Should(gomega.Succeed())
			gomega.Expect(util.DeleteWorkloadsInNamespace(ctx, k8sClient, nsB)).Should(gomega.Succeed())
```

Reply:
Ah, reading this "But I think you right. We're creating jobs in nsB in the next test cases but not in the failed. So looks like kubernetes garbage collector didn't have time to remove workloads and we can't remove clusterqueue because workload exists." it seems the relevant are are actually: order of deletion, and force deletion of workloads.

Reply:
Yes, exactly.